transmart-core-db/grails-app/services/org/transmartproject/db/tree/TreeService.groovy

```diff
@@ -94,32 +90,19 @@ class TreeService implements TreeResource {
         }
     }
 
-    private List<String> getStudyTokens(DbUser user) {
-        List<String> studyTokens = []
-        if (!user.admin) {
-            Collection<Study> studies = accessControlChecks.getDimensionStudiesForUser(user) as Collection<Study>
-            studyTokens = studies.collect { it.secureObjectToken }
-        }
-        studyTokens.sort().unique()
-    }
-
-    private I2b2Secure fetchRootNode(DbUser user, String rootKey) {
+    private I2b2Secure fetchRootNode(User user, String rootKey) {
         if (rootKey == I2b2Secure.ROOT) {
             return null
         }
-        DetachedCriteria criteria = DetachedCriteria.forClass(I2b2Secure)
-                .add(StringUtils.like('fullName', rootKey))
-
-        if (!user.admin) {
-            List<String> tokens = [Study.PUBLIC, 'EXP:PUBLIC'] + getStudyTokens(user)
-            criteria = criteria.add(Restrictions.in('secureObjectToken', tokens))
+        I2b2Secure i2b2Secure = (I2b2Secure) I2b2Secure.createCriteria().get {
+            eq('fullName', rootKey)
         }
-
-        I2b2Secure root = criteria.getExecutableCriteria(sessionFactory.currentSession).uniqueResult() as I2b2Secure
-        if (!root) {
+        assert i2b2Secure : "Node with path ${rootKey} is not found."
+        if (accessControlChecks.canPerform(user, ProtectedOperation.WellKnownOperations.SHOW_SUMMARY_STATISTICS, i2b2Secure)) {
```

Comment:
We actually want to throw the same error when the node does not exist as when the user does not have access.

Reply:
By the way, why change the name from `root` to `i2b2Secure`? Is it more descriptive of what we are fetching?